examples/sync/sync.futil

```diff
@@ -5,38 +5,51 @@ import "primitives/sync.futil";
 // register.
 component main() -> () {
```

Comment:
When you create a new file for this example, can you update the description to talk about what this new program is doing and what the synchronization invariants are i.e., when do the writes blocks and what the expected resolution order for the arbitration is?